### DISCUSSION TITLE
Make SVGs horizontally scrollable in HTML page

### DISCUSSION THREAD

**Discussion by pawamoy** [Textualize/rich#2640] [Q&A] [answered]
I'm using Rich and [Markdown-Exec](https://pawamoy.github.io/markdown-exec/) to add terminal SVGs to my docs dynamically.
The issue is that if I specify a large console width, like 120, then the font size becomes very small because the SVG tries to fit the width of its parent (I think?).

![svgscroll](https://user-images.githubusercontent.com/3999221/201482222-806b1700-062e-479f-a06d-cd2beffc0420.png)

The first SVG has a console width of 150 and the second one 80.

Instead of the SVG fitting the width and reducing font size, I'd like it to always render using the same font size and with an horizontal scroll bar, just like regular code blocks.

Would any of you know how to achieve this :smile:?

The SVG is wrapped in a `div` (I can add an id, or classes, or whatever helps for CSS):
![svgscroll_html](https://user-images.githubusercontent.com/3999221/201482702-3d3f40de-0ce1-484f-ae8e-15b8c3eb5dd0.png)

I've tried setting a larger width on the SVG than the parent div, with `overflow-x: auto` or `overflow-x: scroll` without success :confused: 

Shamefully tagging @squidfunk as maybe you'll have an idea :see_no_evil: :bowing_man:? (I'm using Material for MkDocs)

**Comment by pawamoy** [comment, marked as answer]
Ah, for this exact use-case, I believe `export_html` is the solution: this will be rendered as a code block, with the desired horizontal scrolling. Font size will match the rest of the page.

**Reply by pawamoy** [reply]
Hmmm it seems the style is leaking into the page. I'll open an issue.

**Reply by willmcgugan** [reply]
Try setting `inline_styles=True`

**Reply by pawamoy** [reply]
Both `inline_styles=True` and `inline_styles=False` give the same result.

**Reply by pawamoy** [reply]
Issue opened as https://github.com/Textualize/rich/issues/2641

**Reply by pawamoy** [reply]
I also had to tweak the colors because `blue` and `yellow` looked bad on both light and dark themes: `#7faeff` instead of blue and `#afaf72` instead of yellow. Happy to get better suggestions as I'm bad with colors :laughing:

**Reply by pawamoy** [reply]
Final code:

```python
with open(os.devnull, "w") as devnull:
    console = Console(record=True, width=150, file=devnull)
    console.print(report, markup=True, highlight=False)
print(console.export_html(inline_styles=True, code_format="<pre><code>{code}</code></pre>"))
```
